### PR TITLE
IOS-9433 [iOS]Mistica Implementing the Small style in the Header component

### PR DESCRIPTION
**Pull request by DavidMarinCalleja**
## 🎟️ **Jira ticket**
[IOS-9433](https://jira.tid.es/browse/IOS-9433) [iOS]Mistica Implementing the Small style in the Header component

## 🥅 **What's the goal?**
A new style called Small has been added to the Header component.
[Header Components Specs](https://www.figma.com/file/Xm3r8nRNlbAiZNlkaH5YX1/%F0%9F%94%B8-Header-Component-Specs?type=design&node-id=2930-10098&mode=design&t=WH9HWIDvEpCLrMzi-0)

## 🚧 **How do we do it?**
I've added two new styles to [HeaderViewStyle.swift](https://github.com/Telefonica/mistica-ios/pull/316/files#diff-68d95fd280949cf7dda30acf3168ebb86218f217e816adff7a8fda8a36906d36)
```
public enum HeaderViewStyle {
    case normal
    case normalSmall.    // <-- NEW
    case inverse
    case inverseSmall    // <-- NEW
}

```

## 🧪 **How can I verify this?**
|SwiftUI|UIKit|
|--|--|
|||

|normalSmall|inverseSmall|
|--|--|
| ||

## 🏑 **AppCenter build**
You can use the alpha build https://install.appcenter.ms/orgs/Tuenti-Organization/apps/Mistica-SwiftUI-iOS
Screenshot 2023-10-20 at 10 53 47

```
Version 26.2.0 (1)
Oct 20, 2023 at 10:44
4.71 MB
```

### CODE REVIEW

File: Tests/MisticaSwiftUITests/UI/__Snapshots__/HeaderTests/testFullHeaderAndSmallStyle.1.png
Comment (DavidMarinCalleja) [inline Tests/MisticaSwiftUITests/UI/__Snapshots__/HeaderTests/testFullHeaderAndSmallStyle.1.png:1]: https://www.figma.com/file/Xm3r8nRNlbAiZNlkaH5YX1/%F0%9F%94%B8-Header-Component-Specs?type=design&node-id=0-1&mode=design&t=WH9HWIDvEpCLrMzi-0

File: Tests/MisticaSwiftUITests/UI/__Snapshots__/HeaderTests/testFullHeaderWithInverseAndSmallStyle.1.png
Comment (DavidMarinCalleja) [inline Tests/MisticaSwiftUITests/UI/__Snapshots__/HeaderTests/testFullHeaderWithInverseAndSmallStyle.1.png:1]: https://www.figma.com/file/Xm3r8nRNlbAiZNlkaH5YX1/%F0%9F%94%B8-Header-Component-Specs?type=design&node-id=0-1&mode=design&t=WH9HWIDvEpCLrMzi-0

**Review by WanaldinoTelefonica — APPROVED** [review]
🎆

**Review by MontseTelefonica — APPROVED** [review]

File: MisticaCatalog/Source/Catalog/Mistica/Components/UICatalogHeaderViewController.swift [hunk 1]
```diff
@@ -174,23 +181,37 @@ extension UICatalogHeaderViewController: UITableViewDataSource, UITableViewDeleg
...
         vc.headerView.descriptionValue = HeaderText(text: descriptionText?.text ?? "")
-        vc.headerView.style = style
 
         show(vc, sender: self)
     }
 }
 
 private extension UICatalogHeaderViewController {
-    func createSegmentedControl(reuseIdentifier: String, firstSegmentedTitle: String, secondSegmentedTitle: String) -> UISegmentedControlTableViewCell {
+    func createSegmentedControl(reuseIdentifier: String, segments: String...) -> UISegmentedControlTableViewCell {
```
Comment (dhidalgofadrique) [inline MisticaCatalog/Source/Catalog/Mistica/Components/UICatalogHeaderViewController.swift:209]: ❤️

**Review by dhidalgofadrique — APPROVED** [review]
👍

**Pull request merged**